src/Groups/cosets.jl

```diff
@@ -640,7 +671,7 @@ function intersect(V::AbstractVector{Union{<: GAPGroup, GroupCoset, GroupDoubleC
    else
       G = V[1].G
    end
-   l = GAP.Obj([v.X for v in V])
+   l = GAP.Obj([GapObj(v) for v in V])
```

Comment:
Would this also work (untested)?
```suggestion
   l = GAP.Obj(V; recursive = true)
```